### DISCUSSION TITLE
Can't sync Facebook calendar due to "event without start time"

### DISCUSSION THREAD

**Discussion by tbsv** [bitfireAT/icsx5#39] [Q&A]
Hello,

I'm using ICSx5 for a while without issues. Since a few weeks, I can't sync Facebook calendar anymore. It looks like there is one event without start time that results in the following error.

Error message:
`Event without start time at.bitfire.ical4android.InvalidCalendarException: Event without start time at at.bitfire.ical4android.Event$Companion.fromVEvent(Event.kt:184) at at.bitfire.ical4android.Event$Companion.eventsFromReader(Event.kt:129) at at.bitfire.ical4android.Event$Companion.eventsFromReader$default(Event.kt:80) at at.bitfire.icsdroid.ProcessEventsTask$processEvents$downloader$1.onSuccess(ProcessEventsTask.kt:67) at at.bitfire.icsdroid.CalendarFetcher.fetchNetwork$icsx5_59_2_0_1_gplayRelease(CalendarFetcher.kt:155) at at.bitfire.icsdroid.CalendarFetcher.onRedirect(CalendarFetcher.kt:83) at at.bitfire.icsdroid.CalendarFetcher.fetchNetwork$icsx5_59_2_0_1_gplayRelease(CalendarFetcher.kt:171) at at.bitfire.icsdroid.CalendarFetcher.run(CalendarFetcher.kt:46) at at.bitfire.icsdroid.ProcessEventsTask.processEvents(ProcessEventsTask.kt:106) at at.bitfire.icsdroid.ProcessEventsTask.run(ProcessEventsTask.kt:39) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) at java.lang.Thread.run(Thread.java:923)`

Versions:
ICSx5 2.0.1-gplay
Android 11

Until now, I was not able to identify the event. Do you have an idea how to handle this?

**Comment by devvv4ever** [comment]
Hello tbsv, 

please download the VEVENT and post it here - or if it has sensitive personal data please send it to us via email: play@bitfire.at. We will have a look at it.

Thank you!

**Reply by tbsv** [reply]
I also checked the VEVENT file locally, but all DTSTART timestamps where set.
Meanwhile I figured out an alternative way to import this in my Nextcloud instance. Now, I'm able to synchronize the calendar via CalDav (DAVX5) and the problem with ICSX5 is gone.

**Reply by mantaslu** [reply]
> Meanwhile I figured out an alternative way to import this in my Nextcloud instance. Now, I'm able to synchronize the calendar via CalDav (DAVX5) and the problem with ICSX5 is gone.

This did not work for me. I can add Facebook events as "New subscription from link" in NextCloud, and they show up just fine there. However, in Android, in DAVX5 this calendar appears disabled under WEBCAL (not CALDAV), and if I try to enable it, ICSx5 is invoked with the same original Facebook URL and I am back to square one.

**Reply by devvv4ever** [reply]
Which ICSx5 version did you use? Did you update to ICSx5 2.0.3 already?

**Reply by mantaslu** [reply]
> Which ICSx5 version did you use? Did you update to ICSx5 2.0.3 already?

Yes, 2.0.3.

**Comment by sunkup** [comment]
After analyzing the problem further, we found that facebook does not handle double-quotes (`"`) in organizer names, as defined in the iCal standard supported by us. Facebook uses backslashes to escape double-quotes, which has the parser fail and not understand the following property, which happens to be the start time `DTSTART`. Nextcloud may gracefully handle the issue in some way, making it syncable. 

You can have a look in your VEVENT and see whether there is a double-quote in the common-name parameter of the organizer property, for example like so: 
```
[...]
ORGANIZER;CN=XXXX \"XXXX\":MAILTO:noreply@facebookmail.com
[...]
```

As a work-around you may try to find the facebook setting which sets the the organizer name and change it to not contain double-quotes.

**Reply by mantaslu** [reply]
Thank you for looking into this and for the excellent work. This indeed appears to be the problem. 

> As a work-around you may try to find the facebook setting which sets the the organizer name and change it to not contain double-quotes.

There is no such setting in FB, or any setting related to event exporting. 

I can only add or sync the FB calendar if I unsubscribe on FB from all the events organized by entities having double quotes in their names. There still seems to be no way around it. 

Is there a chance of accommodating this FB bug, even if FB is in the wrong here? 
The FB events show up fine in NextCloud or Thunderbird.

**Reply by loziniak** [reply]
Perhaps you could add an option to gracefully work-around the Facebook bug?

**Reply by sunkup** [reply]
Our focus lies with refactoring and stability improvements at the moment. You are however welcome to create a PR.
